lib/compile.js

```diff
@@ -1,23 +1,89 @@
 'use strict'
 
-var babel = require('babel-core')
-var React = require('react')
-var ReactDOMServer = require('react-dom/server')
-var reval = require('eval')
-require('babel-core/register')
+const babel = require('@babel/core')
+const React = require('react')
+const ReactDOMServer = require('react-dom/server')
+const reval = require('eval')
+const register = require('@babel/register')
 
-module.exports = function compile (data) {
-  var js = babel.transform(data.text, { filename: data.path })
-  var Component = reval(js.code, data.path, null, true)
+/**
+ * @typedef {import("hexo").extend.RendererData} RendererData data of renderer
+ *
+ * @typedef {object} Locals hexo locals such as config, layout, etc
+ *
+ * @typedef {object} Config renderer config
+ * @property {import("@babel/core").TransformOptions} babel babel config
+ *
+ * @typedef {import("react").Component} Component react component
+ */
 
-  return function (locals) {
-    var element = React.createElement(Component.default || Component, locals)
-    var markup = ReactDOMServer.renderToStaticMarkup(element)
+/**
+ * compile text into react component
+ *
+ * @param {RendererData} data data of renderer
+ * @param {Config} config renderer config
+ */
+const compile = (data, config) => {
+  const js = babel.transform(data.text, {
+    filename: data.path,
+    ...config.babel
+  })
+  return reval(js.code, data.path, null, true)
+}
+
+/**
+ * render react component into html
+ *
+ * @param {Component} component react component
+ * @param {Locals} locals hexo locals such as config, layout, etc
+ * @returns
+ */
+const render = (component, locals) => {
+  const element = React.createElement(component.default || component, locals)
+  let markup = ReactDOMServer.renderToStaticMarkup(element)
+
+  if (!locals.layout && markup.toLowerCase().startsWith('<html')) {
+    markup = '<!doctype html>\n' + markup
+  }
+
+  return markup
+}
 
-    if (markup.slice(0, 5).toLowerCase() === '<html') {
-      markup = '<!doctype html>\n' + markup
-    }
+/**
+ * get renderer for register
+ *
+ * @param {Config} config renderer config
+ */
+const getRenderer = config => {
+  /**
+   * @link https://babeljs.io/docs/en/babel-register
+   */
+  register(config.babel)
 
-    return markup
+  /**
+   * @param {RendererData} data data of renderer
+   * @param {Locals} locals hexo locals such as config, layout, etc
+   * @link https://github.com/hexojs/hexo/blob/6.2.0/lib/theme/view.js#L39
+   */
+  const renderer = (data, locals) => {
```

Comment:
According to the [source code](https://github.com/hexojs/hexo/blob/6.2.0/lib/theme/view.js#L129), the `renderer` function does not require any implementation, only a `compile` property is required.